Move performance to java agent, merge javadoc into API page (#5590)
open-telemetry/opentelemetry.io: content/en/docs/languages/java/api.md

  `TracerProvider`, `MeterProvider`, `LoggerProvider`, `ContextPropagators`)
  which is convenient to pass to instrumentation.

The API is designed to support multiple implementations. Two implementations are
provided by OpenTelemetry:

- [SDK](../sdk/) reference implementation. This is the right choice for most
  users.
- [Noop](#noop-implementation) implementation. A minimalist, zero-dependency
  implementation for instrumentations to use by default when the user doesn't
  install an instance.

The API is designed to be taken as a direct dependency by libraries, frameworks,
and application owners. It comes with
[strong backwards compatibility guarantees](https://github.com/open-telemetry/opentelemetry-java/blob/main/VERSIONING.md#compatibility-requirements),
zero transitive dependencies, and
[supports Java 8+](https://github.com/open-telemetry/opentelemetry-java/blob/main/VERSIONING.md#language-version-compatibility).
Libraries and frameworks should depend only on the API and only call methods
from the API, and instruct applications / end users to add a dependency on the
SDK and install a configured instance.

> {{% alert %}} For an exhuastive javadoc reference of all OpenTelemetry Java

Reviewer: How about just "the" instead of "an exhuastive"?

```suggestion
{{% alert title=Javadoc %}} For the Javadoc reference of all OpenTelemetry Java
```

> components, see [javadoc.io](https://javadoc.io/doc/io.opentelemetry).

Reviewer: ```suggestion
components, see [javadoc.io/doc/io.opentelemetry(https://javadoc.io/doc/io.opentelemetry).
```
